pkg/envoy/registry/announcement_handlers.go

```diff
@@ -8,9 +8,9 @@ import (
 	"github.com/openservicemesh/osm/pkg/kubernetes/events"
 )
 
-// releaseCertificateHandler releases certificates based on podDelete events
+// ReleaseCertificateHandler releases certificates based on podDelete events
 // returns a stop channel which can be used to stop the inner handler
-func (mc *MeshCatalog) releaseCertificateHandler() chan struct{} {
+func (mc *ProxyRegistry) ReleaseCertificateHandler(certManager certificate.Manager) chan struct{} {
```

Comment:
@nojnhuh I can see this `ReleaseCertificateHandler` being carved out at some point as a standalone function.

A tiny nit here - the `mc` could be `pr` for ProxyRegistry -- though I appreciate how clean this look in diff view - easy to spot the changes.

We can continue to work on `ReleaseCertificateHandler()` (if we agree it needs more refactoring) in a follow-up PR.

Reply:
Ah yep, changed the other spots but missed this one. Fixed.